pravega-client-examples/src/main/java/io/pravega/example/secure/SecureBatchReader.java

```diff
@@ -0,0 +1,319 @@
+import io.pravega.client.stream.impl.ControllerImpl;
+import io.pravega.client.stream.impl.ControllerImplConfig;
+import io.pravega.client.stream.impl.JavaSerializer;
+import io.pravega.shared.watermarks.Watermark;
+import io.pravega.client.watermark.WatermarkSerializer;
+import io.pravega.shared.NameUtils;
+import io.pravega.client.stream.Stream;
+import io.pravega.client.stream.StreamCut;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.client.segment.impl.Segment;
+import io.pravega.client.stream.Position;
+import io.pravega.client.stream.Stream;
+import io.pravega.client.stream.StreamCut;
+import io.pravega.client.stream.TimeWindow;
+import io.pravega.client.segment.impl.Segment;
+import io.pravega.client.stream.impl.StreamCutImpl;
+import io.pravega.client.stream.TruncatedDataException;
+import org.apache.commons.lang3.ObjectUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Supplier;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.Random;
+
+
+
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import java.net.URI;
+import java.util.UUID;
+import static io.pravega.common.concurrent.ExecutorServiceHelpers.newScheduledThreadPool;
+
+/**
+ * This class demonstrates how to configure a Pravega reader application for:
+ *
+ *    a) SSL/TLS (HTTPS) communications with a Pravega cluster for data-in-transit encryption and
+ *       server authentication.
+ *    b) Passing credentials to a Pravega cluster for authentication and authorization of clients.
+ *
+ * This example can be driven interactively against a running Pravega cluster configured to communicate using SSL/TLS
+ * and "auth" (authentication and authorization) turned on.
+ */
+public class SecureBatchReader {
+
+    private final String scope;
+    private final String stream;
+    private final URI controllerURI;
+
+    // TLS related config
+    private final String truststorePath;
+    private final boolean validateHostName;
+
+    // Auth related config
+    private final String username;
+    private final String password;
+    private ClientConfig clientConfig;
+
+    private static Logger LOG = LoggerFactory.getLogger(SecureBatchReader.class);
+    private static final int THREAD_POOL_SIZE = 5;
+    private static ScheduledExecutorService streamCutLoggerService = newScheduledThreadPool(THREAD_POOL_SIZE, String.format("EventsCount"));
+    private ScheduledExecutorService executor;
```

Comment:
This needs to be set in the constructor and you need a `close()` method on this class (implement `AutoCloseable`) to properly shut it down.

Reply:
Yes Andre